primo/utils/clustering_utils.py

```diff
@@ -138,10 +145,47 @@ def perform_clustering(wd: WellData, distance_threshold: float = 10.0):
         distance_threshold=distance_threshold,
     ).fit(distance_metric)
 
-    wd.data["Clusters"] = clustered_data.labels_
-    # Uncomment the line below to convert labels to strings. Keeping them as
-    # integers for convenience.
-    # wd.data["Clusters"] = "Cluster " + wd.data["Clusters"].astype(str)
-    wd.col_names.register_new_columns({"cluster": "Clusters"})
+    wd.add_new_column_ordered("cluster", "Clusters", clustered_data.labels_)
 
     return clustered_data.n_clusters_
+
+
+def get_pairwise_metrics(wd: WellData, list_wells: list) -> pd.DataFrame:
+    """
+    Returns pairwise metric values for all well pairs in each clusters
+
+    Parameters
+    ----------
+    wd : WellData
+        Object containing well data
+
+    list_wells : list
+        List of wells for which pairwise metrics are needed to
+        be calculated
+
+    Returns
+    -------
+    pairwise_metrics : DataFrame
+        DataFrame containing the pairwise metric values
+    """
+    well_pairs = list(combinations(list_wells, 2))
+    pairwise_metrics = pd.DataFrame()
+
+    # Compute pairwise distances
+    pairwise_metrics["distance"] = distance_matrix(
+        wd, {"distance": 1}, list_wells
+    ).stack()[well_pairs]
+
+    # Compute pairwise age range
+    if wd.column_names.age is not None:
+        pairwise_metrics["age_range"] = distance_matrix(
+            wd, {"age": 1}, list_wells
```

Comment:
Do we foresee a possibility where these weights for age and depth need to be user defined arguments? If yes, recommend moving them into parameters now.

Reply:
In this function, we are computing distance, age range, and depth range independently, as they required for efficiency calculations. In other words, the weights are fixed in this case, so I think it is not needed to make them user-defined arguments. 

We may need to make them user-defined arguments for the clustering step (`perform_clustering` function). But, we will do that after we figure out the correct way to scale all the three metrics and take the weighted sum.

Reply:
Alright.